processing/src/main/java/io/druid/query/topn/TopNMapFn.java

```diff
@@ -42,38 +42,28 @@
         return LONG_TRANSFORMER;
       case FLOAT:
         return FLOAT_TRANSFORMER;
+      case DOUBLE:
+        return DOUBLE_TRANSFORMER;
       default:
         throw new IAE("invalid type: %s", outputType);
     }
   }
 
-  private static Function<Object, Object> STRING_TRANSFORMER = new Function<Object, Object>()
-  {
-    @Override
-    public Object apply(Object input)
-    {
-      return Objects.toString(input, null);
-    }
+  private static Function<Object, Object> STRING_TRANSFORMER = input -> Objects.toString(input, null);
+
+  private static Function<Object, Object> LONG_TRANSFORMER = input -> {
+    final Long longVal = DimensionHandlerUtils.convertObjectToLong(input);
+    return longVal == null ? 0L : longVal;
```

Comment:
Maybe make sense to extract this 0L as `Long` constant, to ensure boxing doesn't create a different object each time.